File: app/src/main/java/com/vasthread/sleeptile/SleepTileService.java
package com.vasthread.sleeptile;

import android.service.quicksettings.TileService;

import java.io.OutputStream;

public class SleepTileService extends TileService {

    @Override
    public void onClick() {
        try {
            Process process = Runtime.getRuntime().exec("su");
            try (OutputStream output = process.getOutputStream()) {
                output.write("input keyevent 26\n".getBytes());
                output.write("exit\n".getBytes());
            }
            process.waitFor();
        } catch (Exception e) {
            e.printStackTrace();
        }
    }
}

File: app/src/main/java/com/vasthread/sleeptile/SleepTileService.kt
package com.vasthread.sleeptile

import android.service.quicksettings.TileService

class SleepTileService : TileService() {

    override fun onClick() {
        super.onClick()
        runCatching {
            val process = Runtime.getRuntime().exec("su")
            process.outputStream.use {
                it.write("input keyevent 26\n".toByteArray())
                it.write("exit\n".toByteArray())
            }
            process.waitFor()
        }
    }
}